File: be/src/project-members/dto/create-project-member.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsEmail, IsEnum, IsNotEmpty, IsOptional } from 'class-validator';

export class CreateProjectMemberDto {
  @ApiProperty({ example: 'example@example.com' })
  
  @IsNotEmpty()
  email: string;

  @ApiProperty({ example: 'viewer' })
  @IsOptional()
  @IsEnum(['viewer', 'editor', 'leader'])
  role: 'viewer' | 'editor' | 'leader';
}

File: backend-java/src/main/java/com/qlda/backend/projectmember/dto/CreateProjectMemberDto.java
package com.qlda.backend.projectmember.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.Data;

@Data
public class CreateProjectMemberDto {
    @NotBlank(message = "Email không được để trống")
    @Email(message = "Email không hợp lệ")
    private String email;

    @Pattern(regexp = "viewer|editor|leader", message = "Role phải là viewer, editor hoặc leader")
    private String role = "viewer";
}

